### DISCUSSION TITLE
Chart Or work Flow drawing

### DISCUSSION THREAD

**Discussion by peuportier** [open-webui/open-webui#2545] [General]
Hello, Dear Open-Webui Community,

First of all, thank you for the great work you are doing here. I truly appreciate the dedication and effort put into developing this amazing tool.

I have a suggestion that I believe could enhance the functionality of Open-Webui and provide even greater value to its users. Is it possible for Open-Webui to have the ability to draw various types of charts in the future? For instance, it would be incredibly useful if we could generate pie charts, bar diagrams, flow charts, and more through simple text commands.

One potential integration that comes to mind is using Mermaid, a versatile diagramming and charting tool. Imagine being able to type commands like "draw me a pie chart for the percentage market share of PlayStation and Nintendo in Asia" and getting an immediate, visually appealing chart in response. This feature could significantly boost the usability and versatility of Open-Webui for various data visualization needs.

I came across a GitHub project where a developer has been working on something similar, and it got me thinking about the potential integration of such capabilities into Open-Webui. I believe this would be a fantastic addition and would love to hear your thoughts on the feasibility of implementing this feature.

[](https://github.com/fraserxu/diagram-gpt)
[](https://github.com/nilooy/flowgpt)

Thank you once again for your continuous hard work and dedication.

**Comment by GaelCMora** [comment]
Hello.

I have the same suggestion: if we work in a company, we can view our data in **charts** and share them to create a report.

In fact, some features on the OpenWebUI website aren't compatible with **OpenWebUI**, but if we implement them directly in the instance, our work becomes more efficient and productive.
